File: discord/src/main/java/io/github/shaksternano/borgar/command/ServerCountCommand.java
package io.github.shaksternano.borgar.command;

import com.google.common.collect.ListMultimap;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;

import java.util.List;

public class ServerCountCommand extends SimpleCommand {

    /**
     * Creates a new command object.
     *
     * @param name        The name of the command. When a user sends a message starting with {@link Command#PREFIX}
     *                    followed by this name, the command will be executed.
     * @param description The description of the command. This is displayed in the help command.
     */
    public ServerCountCommand(String name, String description) {
        super(name, description);
    }

    @Override
    protected String response(List<String> arguments, ListMultimap<String, String> extraArguments, MessageReceivedEvent event) {
        var serverCount = event.getJDA().getGuildCache().size();
        var message = "This bot is in " + serverCount + " server";
        if (serverCount != 1) {
            message += "s";
        }
        return message + ".";
    }
}

File: chat/src/main/kotlin/io/github/shaksternano/borgar/chat/command/ServerCountCommand.kt
package io.github.shaksternano.borgar.chat.command

import io.github.shaksternano.borgar.chat.event.CommandEvent
import io.github.shaksternano.borgar.core.util.asSingletonList

object ServerCountCommand : NonChainableCommand() {

    override val name: String = "servercount"
    override val aliases: Set<String> = setOf("servers")
    override val description: String = "Gets the number of servers that this bot is in."

    override suspend fun run(arguments: CommandArguments, event: CommandEvent): List<CommandResponse> {
        val guildCount = event.manager.getGuildCount()
        var message = "This bot is in $guildCount server"
        if (guildCount != 1) {
            message += "s"
        }
        return CommandResponse(message).asSingletonList()
    }
}
